specs/~dynamic-names.yml

```
@@ -0,0 +1,358 @@
+overview: |
+  Rationale: this special notation was introduced primarily to allow the dynamic
+  loading of partials. The main advantage that this notation offers is to allow
+  dynamic loading of partials, which is particularly useful in cases where
+  polymorphic data needs to be rendered in different ways. Such cases would
+  otherwise be possible to render only with solutions that are convoluted,
+  inefficient, or both.
+  Example.
```

I suggest adding blank lines in even more places.

```suggestion
  inefficient, or both.

  Example.
```

---

Fine by me!